ee/hogai/graph/taxonomy/toolkit.py

```diff
@@ -284,45 +284,85 @@ def _get_custom_tools(self) -> list:
         """Get custom tools. Override in subclasses to add custom tools."""
         raise NotImplementedError("_get_custom_tools must be implemented in subclasses")
 
-    def retrieve_entity_properties(self, entity: str, max_properties: int = 500) -> str:
+    def retrieve_entity_properties(self, entity: str | list[str], max_properties: int = 500) -> dict[str, str]:
         """
-        Retrieve properties for an entitiy like person, session, or one of the groups.
+        Retrieve properties for an entity or multiple entities like person, session, or one of the groups.
         """
-
-        if entity not in ("person", "session", *[group.group_type for group in self._groups]):
-            return TaxonomyErrorMessages.entity_not_found(entity, self._entity_names)
-
-        if entity == "person":
-            qs = PropertyDefinition.objects.filter(team=self._team, type=PropertyDefinition.Type.PERSON).values_list(
-                "name", "property_type"
-            )
-            props = self._enrich_props_with_descriptions("person", qs)
-        elif entity == "session":
-            # Session properties are not in the DB.
-            props = self._enrich_props_with_descriptions(
-                "session",
-                [
+        if isinstance(entity, str):
+            return self._bulk_get_entity_properties([entity], max_properties)
+        else:
+            return self._bulk_get_entity_properties(entity, max_properties)
+
+    def _bulk_get_entity_properties(self, entities: list[str], max_properties: int = 500) -> dict[str, str]:
+        result = {}
+        group_entities = []
+        entities_set = set(entities)
+        for entity in entities_set:
+            if entity not in ("person", "session", *[group.group_type for group in self._groups]):
+                result[entity] = TaxonomyErrorMessages.entity_not_found(entity, self._entity_names)
+                continue
+            if entity == "person":
+                person_qs = PropertyDefinition.objects.filter(
+                    team=self._team, type=PropertyDefinition.Type.PERSON
+                ).values_list("name", "property_type")
+                result[entity] = (
+                    self._format_properties(self._enrich_props_with_descriptions("person", person_qs))
```

Comment:
Just for me to understand, why are we using `_format_properties` but we were not using it on master?